packages/store/src/external/memory.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2019-20 Agoric, under Apache license 2.0
+
+// @ts-check
+import { makeWeakStore } from '../weak-store';
+import '../types';
+
+/**
+ * Create a completely in-memory "external" store.  This store will be
+ * garbage-collected in the usual way, but it will not page out any objects to
+ * secondary storage.
+ *
+ * @template {(...args: any[]) => ExternalInstance} M
+ * @param {string} instanceName
+ * @param {M} maker
+ * @returns {ExternalStore<M>}
+ */
+export function makeMemoryExternalStore(instanceName, maker) {
+  return harden({
+    makeInstance: maker,
```

Comment:
This makes beautifully clear that the external store really is just a logical extension of the heap, rather than something that visibly has collection-like properties. Here, there is no collection of made instances at all. They're just stored in the language's heap of objects. End of story. Beautiful!